pm_icecon/bt/compute_bt_ic.py

```diff
@@ -636,84 +636,76 @@ def coastal_fix(
 
     # Note: some of these conditional arrays might be set more than 1x
 
-    # Compute shifted conc grid, for land check
-    is_rolled_land = np.roll(land_mask, (0, 1), axis=(1, 0))  # land check
-
-    # For offp1 of [0, 1], the rolls are:
-    tip1jp1 = np.roll(temp, (-1, -1), axis=(1, 0))
-    tim1jp1 = np.roll(temp, (1, -1), axis=(1, 0))
-    tip1jp0 = np.roll(temp, (-1, 0), axis=(1, 0))
-    tim1jp0 = np.roll(temp, (1, 0), axis=(1, 0))
-
-    tip0jp1 = np.roll(temp, (0, -1), axis=(1, 0))
-
-    is_temp0 = temp == 0
-    is_considered = is_temp0 & is_rolled_land
-
-    is_tip1jp1_lt0 = tip1jp1 <= 0
-    is_tim1jp1_lt0 = tim1jp1 <= 0
-    is_tip1jp0_lt0 = tip1jp0 <= 0
-    is_tim1jp0_lt0 = tim1jp0 <= 0
-
-    is_tip0jp1_eq0 = tip0jp1 == 0
+    def _conc_change124(*, shifts, land_mask, temp, conc2):
```

Comment:
I realize this isn't a great implementation, and there's probably a better way to express what shifts we want to happen in each of these operations, but this made it clearer for me where logic was being reused.